float64[] rvec
float64[] tvec
